Formula/newrelic-infra-agent.rb

```diff
@@ -21,8 +21,7 @@ class NewrelicInfraAgent < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux: "ef8260a5a4659a6560e9530513d7a3e99ec2381c1028a8d5c68d27fc3909da48"
   end
 
-  # https://github.com/newrelic/infrastructure-agent/issues/723
-  depends_on "go@1.16" => :build
+  depends_on "go@1.18" => :build
```

Comment:
Go version needs to be pinned and it will be updated once we bump the Go version in the agent.

Reply:
That's not really how Homebrew works. If it works with the latest version it must use the unversioned dependency. If Go is updated and it breaks someone will pin the version to the last Go release.

Reply:
I find it not really practical to wait until something is broken to be able to pin it. We test really carefully the Go 
 version update in the infra agent before releasing it and therefore we would like to have that opportunity with brew too. This is the only way I can think of, pin it to the latest supported version and after the agent is carefully tested then update it in the formula as we do with the rest of supported operative systems.

I don't understand why having Go version pinned in the formula can be a problem. I can only see that it makes it more reliable, but I might be missing something.

Reply:
That's the approach that Debian and Red Hat take. But it's not the approach Homebrew takes. Everything must use the latest available dependencies. What upstream wants is not really relevant since we're talking about open source software here that anyone can distribute.

Reply:
I can understand that this is the desired scenario. And maybe it works for other software. In our case it's not feasible, so I prefer to have the pinned version and update it as soon as we bump Go version (as we did now) over having the latest unpinned version and have a broken release.

Can we make a [exception](https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Formulae.md#sometimes-there-are-exceptions) here?

Reply:
Not on my account since so far that has meant that the Go version it uses barely avoids deprecation (and therefore disabling of the formula). If upstream does not test with recent Go versions it's not up to Homebrew to accommodate that with exceptions that are completely opposed to the usual rules.

Reply:
As I mention updating Go or any dependency is a process we take really seriously as it can impact the final users, so we don't jump into the latest version as soon as it is released. Instead we plan and test it before bumping. 

We think that not having the pinned version can lead to a broken package while not having any benefit for the users.

Reply:
You can maintain a tap where you take that approach and homebrew will continue the approach that users have come to expect from us.

Reply:
We discussed it and I have unpinned Go version https://github.com/Homebrew/homebrew-core/pull/101909/commits/e887577f90fa44794616c406e954b46b32bcbeb4